treeherder/model/models.py

```diff
@@ -817,6 +818,26 @@ def update_autoclassification_bug(self, bug_number):
             return classification.set_bug(bug_number)
 
 
+class TaskclusterMetadata(models.Model):
```

Comment:
I think the main advantages here are:
* Faster schema changes for both jobs and taskcluster metadata where required (since there will be less data to rewrite)
* Better logical separation of concerns in both the database and application code (this is somewhat subjective but I'll make the claim here).

Disadvantages:
* Minor cost of a database join (mitigated by using indexes on the foreign key relationship)